toolkit/docs/CVE/CVE-Quickstart-Guide.md

```diff
@@ -0,0 +1,167 @@
+
+---
+
+### **4. Apply Existing Patches**
+
+1. Identify existing patches in the `<pkg>.spec` file (e.g., `Patch0`, `Patch1`).
+2. Apply patches sequentially starting from `Patch0`:
+   ```bash
+   patch -p1 --fuzz=0 < ~/azurelinux/SPECS/<pkg>/CVE-xxxx-yyyy.patch
+   ```
+3. Commit the changes:
+   ```bash
+   git add -u
+   git commit -m "In pace with azl"
+   ```
+
+---
+
+### **5. Apply the New CVE Patch**
+
+1. Obtain the CVE patch:
+   - Search on NIST ([NVD](https://nvd.nist.gov/vuln/detail/CVE-xxxx-yyyy)) or GitHub for the patch.
+   - Download and rename it:
+     ```bash
+     wget https://github.com/<pkg>.patch
+     mv <patch-name>.patch CVE-xxxx-yyyy.patch
+     ```
+2. Apply the patch:
+   ```bash
+   patch -p1 --fuzz=0 < CVE-xxxx-yyyy.patch
+   ```
+3. Verify and commit:
+   ```bash
+   git diff
+   git add -u
+   git commit -m "Fix CVE patch"
+   ```
+4. Create a new patch file:
+   ```bash
+   git log --oneline
+   git format-patch -1 HEAD
+   mv <patch-name>.patch CVE-xxxx-yyyy.patch
+   ```
+
+---
+
+### **6. Update the Repository**
+
+1. Create a new branch for your azlinux repository:
+   ```bash
+   git checkout -b <alias/CVE-xxxx-yyyy>
+   ```
+2. Copy the newly created patch to the `SPECS` directory.
+3. Update the `<pkg>.spec` file:
+   - Increment the release number.
+   - Add the patch under `PatchX`.
+   - Update the changelog:
+     ```plaintext
+     * Mon Dec 02 2024 User-Name <alias@microsoft.com> -<pkg-version>-<release>
+     - Fix CVE-xxxx-yyyy with an upstream patch
+     ```
+4. Commit and push changes:
+   ```bash
+   git add -u
+   git commit -m "Fix CVE-xxxx-yyyy"
+   git push --set-upstream origin <alias/CVE-xxxx-yyyy>
+   ```
+
+---
+
+### **7. Test the Fix Locally**
+
+- **Build for 2.0**:
+  ```bash
+  git checkout 2.0-stable -- resources/
+  sudo make build-packages REBUILD_TOOLS=y SOURCE_URL="https://azurelinuxsrcstorage.blob.core.windows.net/sources/core"  PACKAGE_BUILD_LIST="<pkg-name>" PACKAGE_REBUILD_LIST="<pkg-name>" SRPM_PACK_LIST="<pkg-name>" RUN_CHECK=y SRPM_FILE_SIGNATURE_HANDLING=update
+  ```
+- **Build for 3.0**:
+  ```bash
+  cd /azurelinux/toolkit
+  ./pkgbld.sh -c -p "<package-name>"
+  ```
+
+---
+
+### **8. Create and Submit a PR**
+
+1. Create a PR in the [Azure Linux GitHub repository](https://github.com/microsoft/azurelinux):
+   - Title: `Patch <pkg> for CVE-xxxx-yyyy`.
+   - Include CVE details and NIST link in the description.
+2. Set the base branch:
+   - `fasttrack/3.0` for 3.0.
+   - `fasttrack/2.0` for 2.0.
+
+---
+
+### **9. Perform Buddy Build**
+
+1. Navigate to [ADO Buddy Build](https://dev.azure.com/mariner-org/mariner/_build?definitionId=2190&_a=summary).
+2. Run a new pipeline:
+   - Use `PR-<GitHub-PR-number>` as the spec.
+   - Add `<pkg-name>` for the build.
+3. Paste the Buddy Build link in the PR.
+
+---
+
+### **10. Complete the Process**
+
+- Ensure the PR is reviewed, approved, and merged.
+- Monitor for additional verification or testing needs.
+
+---
+
```

Comment:
Any specific section needed for Dispute?
https://dev.azure.com/mariner-org/mariner/_wiki/wikis/mariner.wiki/1688/Resolve-and-Dispute-Process